implementations/rust/ockam/ockam_api/src/session/sessions.rs

```diff
@@ -21,24 +22,38 @@ pub type Replacer = Box<dyn FnMut(Route) -> Replacement + Send>;
 pub struct Session {
     key: String,
     ping_route: Route,
-    status: Status,
+    status: ConnectionStatus,
     replace: Replacer,
     pings: Vec<Ping>,
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum Status {
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Encode, Decode, Serialize, Deserialize)]
+pub enum ConnectionStatus {
+    #[n(0)]
     Down,
+    #[n(1)]
     Degraded,
+    #[n(2)]
     Up,
 }
 
-impl fmt::Display for Status {
+impl fmt::Display for ConnectionStatus {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
-            Status::Down => write!(f, "down"),
-            Status::Degraded => write!(f, "degraded"),
-            Status::Up => write!(f, "up"),
+            ConnectionStatus::Down => write!(f, "down"),
+            ConnectionStatus::Degraded => write!(f, "degraded"),
+            ConnectionStatus::Up => write!(f, "up"),
+        }
+    }
+}
+
+impl From<String> for ConnectionStatus {
+    fn from(value: String) -> Self {
+        match value.as_str() {
+            "down" => ConnectionStatus::Down,
+            "degraded" => ConnectionStatus::Degraded,
+            "up" => ConnectionStatus::Up,
+            _ => ConnectionStatus::Down,
```

Comment:
I'd rather have a `TryFrom` that is failing when the input string is incorrect. Otherwise we might have a silent failure in a bad situation where we just think that the node is currently down.

Reply:
makes perfectly sense! done!